src/Domain/Organization/Services/OrganizationService.php

```diff
@@ -0,0 +1,121 @@
+        private OrganizationRepository $organizationRepository,
+        private OrganizationSavedAddressService $organizationSavedAddressService,
+        private AuthService $authService,
+    ) {}
+
+    /**
+     * @return LengthAwarePaginator<Organization>
+     */
+    public function index(OrganizationIndexDto $dto): LengthAwarePaginator
+    {
+        return $this->organizationRepository->index($dto);
+    }
+
+    public function show(string $id): Organization
+    {
+        return $this->organizationRepository->show($id);
+    }
+
+    public function create(OrganizationCreateDto $dto): Organization
+    {
+        $organization = $this->organizationRepository->create($dto);
+
+        $forceDefault = count($dto->shipping_addresses) === 1;
+        /** @var OrganizationSavedAddressCreateDto $shipping_address */
+        foreach ($dto->shipping_addresses as $shipping_address) {
+            $this->organizationSavedAddressService->storeAddress($shipping_address, SavedAddressType::SHIPPING, $organization->getKey(), $forceDefault);
+        }
+        $organization->refresh();
+
+        return $organization;
+    }
+
+    public function update(Organization $organization, OrganizationUpdateDto $dto): Organization
+    {
+        return $this->organizationRepository->update($organization, $dto);
+    }
+
+    public function delete(string $id): void
+    {
+        $this->organizationRepository->delete($id);
+    }
+
+    public function register(OrganizationRegisterDto $dto): Organization
+    {
+        return DB::transaction(function () use ($dto): Organization {
+            $organization = $this->organizationRepository->registerOrganization($dto);
+
+            $user = $this->authService->register(RegisterDto::from([
+                'email' => $dto->creator_email,
+                'password' => $dto->creator_password,
+                'name' => $dto->creator_name,
+                'captcha_token' => $dto->captcha_token,
+            ]));
+
+            $organization->users()->attach($user->getKey());
+
+            return $organization;
+        });
+    }
+
+    /**
+     * @throws ClientException
+     */
+    public function myOrganization(): Organization
+    {
+        /** @var User $user */
+        $user = Auth::user();
+
+        $organization = $user->organizations()->first();
+
+        if (!$organization) {
+            throw new ClientException(Exceptions::CLIENT_USER_NOT_IN_ORGANIZATION);
+        }
+
+        return $organization;
+    }
+
+    /**
+     * @throws ClientException
+     */
+    public function myOrganizationEdit(OrganizationPublicUpdateDto $dto): Organization
+    {
+        /** @var User $user */
+        $user = Auth::user();
+
+        $organization = $user->organizations()->first();
+
+        if (!$organization) {
+            throw new ClientException(Exceptions::CLIENT_USER_NOT_IN_ORGANIZATION);
+        }
+
+        return $this->organizationRepository->myUpdate($organization, $dto);
```

Comment:
You could just rewrite this whole method as

```
return $this->organizationRepository->myUpdate($this->myOrganization(), $dto);
```

Reply:
done